test cases/frameworks/1 boost/meson.build

```diff
@@ -9,21 +9,18 @@ add_project_arguments(['-DBOOST_LOG_DYN_LINK'],
 # within one project. The need to be independent of each other.
 # Use one without a library dependency and one with it.
 
-nolinkdep = dependency('boost', modules: 'utility')
```

Comment:
Why was this removed?

Reply:
Semantically there is no difference to depend on simply boost or depend on a "headers only library" which the `utility` module is.

Reply:
Having a dependency that only involves using headers and no libraries is a good test to have. It ensures that our code doesn't assume that a dependency has both libs and includes. Does this not work on some platforms or something else?

Reply:
After my changes to the boost dependency this test is equal to the last test `nomoddep = dependency('boost')`.

Previously you could depend on for example `utility`. But today it will fail because `utility` is not a valid "module" any more. And it is not a valid module any more because it doesn't change anything. You will automatically get access to all the "header-only" boost modules because you only need -I/path/to/boost/include/folder to your build commands.

Reply:
Ah, I see. So it worked on accident earlier?

Reply:
No it was on purpose, it actually checked if for example `boost/utility.h` existed, but I don't think that makes sense because you cannot split up the boost headers. You always have to install all of them because they have very complex dependencies on each other. And it ended up confusing users because sometimes the header file and the library file has the same name and then their dependency could be reported as fine, but didn't actually work, because the header file existed but not the library. It is very common to only install a subset of the libraries.

Reply:
I think the new behaviour makes sense, but I don't know if it is acceptable to break existing build files which might be looking for the `utility` module, and so on. Perhaps it's fine since it's a backwards-compatible change. Let's go with that for now and revisit this if people complain.